### PR TITLE
Fix idSearch

### PR DESCRIPTION
**Pull request by TaYaKi71751**
# Related Issues
- https://github.com/project-violet/violet/issues/311
> - https://github.com/project-violet/violet/pull/310#discussion_r1439095047
> - https://github.com/project-violet/violet/pull/310#discussion_r1439095074

### CODE REVIEW

**Comment by TaYaKi71751** [comment]
https://github.com/project-violet/violet/blob/493817228a821ede12d721da62e9c0c796b197e4/lib/component/hentai.dart#L70C1-L124C4
@violet-dev  idSearch 에서 exhentai 에는 시도하지 않습니다.

File: lib/component/hentai.dart [hunk 1]
```diff
@@ -80,49 +80,87 @@ class HentaiManager {
...
+      'EHash': hash,
+      'Title': articleEh.title,
+      'Artists': articleEh.artist?.join('|') ?? 'N/A',
+    };
+
+    return SearchResult(results: [QueryResult(result: meta)], offset: -1);
+  }
+
+  static Future<SearchResult> idSearchExhentai(String what) async {
+    final id = int.parse(what);
```
Comment (TaYaKi71751) [inline lib/component/hentai.dart:140]: Add exhentai to idSearch

File: lib/component/hentai.dart [hunk 2]
```diff
@@ -80,49 +80,87 @@ class HentaiManager {
...
-          'Artists': articleEh.artist?.join('|') ?? 'N/A',
-        };
-
-        return SearchResult(results: [QueryResult(result: meta)], offset: -1);
+        return await idSearchEhentai(what);
       } catch (e, st) {
         Logger.error('[hentai-idSearch] E: $e\n'
             '$st');
+        try {
+          return await idSearchExhentai(what);
```
Comment (TaYaKi71751) [inline lib/component/hentai.dart:92]: Add exhentai to idSearch

**Review by violet-dev — APPROVED** [review]
감사합니다~!!

**Pull request merged**